### PR TITLE
Python 3 11 support

### PR DESCRIPTION
**Pull request by rpsteiger**
Hello,
I spent some time to make this project work with current Python version 3.11.  

It should work out of the box, like so:
```
python3 -m venv .
source bin/activate
pip3 install -r requirements
python3 app.py
```

or using Docker, like so:
```
docker build -t aoe2-api:v1 .
docker run --rm -d -p 8080:8080 aoe2-api:v1
```

I changed the Dockerfile to use gunicorn, since I think this is the easiest solution that works well in a production environment.

Tested to work in Python3.11. The unit tests also work.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,10 @@
-FROM tiangolo/uwsgi-nginx-flask:python3.7
-COPY . /app
-RUN pip install -r requirements.txt
+FROM python:latest
```
Comment (aalises) [inline Dockerfile:1]: Would it be worth it to pin it to a specific version? e.g. `FROM python:3.12.3` or `FROM python:3`

Using the `latest` tag might cause issues when the underlying version changes and causes some incompatibilities, although it's quite unlikely in this case, to be fair, leave it to your choice

**Review by aalises — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (rpsteiger) [reply Dockerfile:1]: HI @aalises thank you for your message. Yes that would be a good idea. Let me do that.
Btw, I just learned that you cannot use python-slim when you have a panda dependency. That's why I went with the the python base image

Comment (rpsteiger) [reply Dockerfile]: @aalises I changed the`latest` tag 👍 😃 to `3.12.2-bookworm`

**Comment by rpsteiger** [comment]
@aalises If you are happy with it, you can merge it. I don't have the rights necessary :)

**Pull request merged**